clash-lib/src/Clash/Rewrite/Util.hs

```diff
@@ -241,7 +264,7 @@ applyDebug lvl _transformations name exprOld hasChanged exprNew =
     error $ $(curLoc) ++ "Expression changed without notice(" ++ name ++  "): before"
                       ++ before ++ "\nafter:\n" ++ after
 
-  traceIf (lvl >= DebugName && hasChanged) name $
+  traceIf (lvl >= DebugName && hasChanged) (name <> " {" <> show nTrans <> "}") $
```

Comment:
This purposely printed as `{n}` as to not collide with uniques (printed as `[n]`)